pkg/cli/admin/upgrade/upgrade.go

```diff
@@ -247,13 +249,29 @@ func (o *Options) Run() error {
 					break
 				}
 			}
+			if o.AllowNotRecommended {
+				for _, upgrade := range cv.Status.ConditionalUpdates {
+					if c := findCondition(upgrade.Conditions, "Recommended"); c != nil && c.Status != metav1.ConditionTrue {
+						if upgrade.Release.Version == o.To {
+							update = &configv1.Update{
+								Version: upgrade.Release.Version,
+								Image:   upgrade.Release.Image,
+							}
```

Comment:
We should add additional complaining here like:

```console
fmt.Fprintf(o.ErrOut, "warning: --allow-not-recommended is bypassing %s=%s %s: %s\n", c.Type, c.Status, c.Reason, c.Message)

Reply:
Unless we are  waiting for user confirmation to wait before moving to the version the additional complaining does not give any significantly value.

Reply:
I could come up with `fmt.Fprintf(o.ErrOut, "warning: with --allow-not-recommended you have accepted the risk with %s and bypassing %s=%s %s: %s\n", o.To, c.Type, c.Status, c.Reason, c.Message)` which I felt would be more appropriate .

Reply:
fixed https://github.com/openshift/oc/commit/7316d191bb03657921a4680fdb4a12106be33c81